### PR TITLE
Updated viral consensus sequencing tutorial

### PR DESCRIPTION
**Pull request by niemasd**
* Changed `ivar` to `viral_consensus`

### CODE REVIEW

File: src/content/viral-amplicon/steps/Step2.md [hunk 1]
```diff
@@ -5,30 +5,20 @@ import Execute from "$components/Execute.svelte";
 
 First, we'll use `minimap2` to align the sequencing paired-end reads in `reads_R1.fq` and `reads_R2.fq`to the reference genome located at `$REF`. While `minimap2` aligns the reads, we will <Link href="https://en.wikipedia.org/wiki/Pipeline_(Unix)#Pipelines_in_command_line_interfaces">pipe</Link> the resulting SAM stream to `samtools` to sort. Run the following:
```
Comment (robertaboukhalil) [inline src/content/viral-amplicon/steps/Step2.md:6]: Minor typo:
```suggestion
First, we'll use `minimap2` to align the sequencing paired-end reads in `reads_R1.fq` and `reads_R2.fq` to the reference genome located at `$REF`. While `minimap2` aligns the reads, we will <Link href="https://en.wikipedia.org/wiki/Pipeline_(Unix)#Pipelines_in_command_line_interfaces">pipe</Link> the resulting SAM stream to `samtools` to sort. Run the following:
```

File: src/content/viral-amplicon/steps/Step2.md [hunk 2]
```diff
@@ -5,30 +5,20 @@ import Execute from "$components/Execute.svelte";
...
-  - In general, the pipeline syntax `A | B` means "execute command `A`, and pipe whatever it outputs (via standard output) into command `B` (via standard input)"
-  - Piping is useful for many reasons, e.g. minimizing slow disk access, reducing how many intermediate files we need to store (saving space), etc.
-- The `samtools` command we're piping the SAM stream into is the following: `samtools sort -o untrimmed.sorted.bam`
-  - `sort` tells `samtools` that we want to use its sorting functionality
-  - `-o untrimmed.sorted.bam` tells `samtools` that we want to write the output BAM results to the file `untrimmed.sorted.bam` (rather than printing them to standard output)
-    - BAM is a binary (i.e., non-human-readable) and (potentially) compressed file format that stores the same information as the SAM format
-  - Rather than having `samtools` read an input file (by specifying a positional argument at the end of the command), we're having it read the input data from standard input
+After running the above command, we will have successfully mapped our reads (`reads_R1.fq` and `reads_R2.fq`) to the reference genome (`$REF`) and written the results to the file `reads.mapped.bam`.
```
Comment (robertaboukhalil) [inline src/content/viral-amplicon/steps/Step2.md:18]: Minor typo:
```suggestion
After running the above command, we will have successfully mapped our reads (`reads_R1.fq` and `reads_R2.fq`) to the reference genome (`$REF`) and written the results to the file `reads.mapped.sam`.
```

File: src/content/viral-amplicon/steps/Step3.md [hunk 3]
```diff
@@ -3,33 +3,45 @@ import Link from "$components/Link.svelte";
...
+<Execute command="cat consensus.fa" />
+
+This consensus genome FASTA file is the other key result of our amplicon sequencing analysis, and it is what will be used in any downstream analyses (e.g. transmission clustering, phylogenetic inference, lineage assignment, etc.).
+
+We will have also output the nucleotide counts (`position_counts.tsv`) and insertion counts (`insertion_counts.json`), which we can use for quality-checking. Specifically, we can see how many reads supported each nucleotide or insertion in the consensus sequence that we called.
+
+To see the contents of the position counts TSV file, run the following:
 
-After running the above command, we will have successfully trimmed our mapped reads and written the trimmed (no longer sorted) results to the file `trimmed.unsorted.bam`.
+<Execute command="cat position_counts.tsv" />
```
Comment (robertaboukhalil) [inline src/content/viral-amplicon/steps/Step3.md:43]: Looks like that file has ~30K lines so this streams all of it on screen. What do you think about replacing this with a `head` command?

File: src/content/viral-amplicon/steps/Step3.md [hunk 4]
```diff
@@ -3,33 +3,45 @@ import Link from "$components/Link.svelte";
...
 
-To see the first few lines of the (unsorted) trimmed BAM output file in the human-readable SAM format, run the following:
+To see the contents of the insertion counts JSON file, run the following:
 
-<Execute command="samtools view -h trimmed.unsorted.bam | \ head -n 5" />
+<Execute command="cat insertion_counts.json" />
```
Comment (robertaboukhalil) [inline src/content/viral-amplicon/steps/Step3.md:47]: When I run this, I see a JSON where the keys are empty strings. Is that expected, or is that a bug on my end?

```json
  "21447": {
    "": 6
  },
  "21446": {
    "": 6
  },
```

**Review by robertaboukhalil — CHANGES_REQUESTED** [review]
Thank you so much! Just a few minor questions/suggestions

File: src/content/viral-amplicon/steps/Step3.md (hunk 3 above)
Comment (niemasd) [reply src/content/viral-amplicon/steps/Step3.md:43]: Replacing with `head` sounds great! Maybe `head -5 position_counts.tsv`? Can you make the change?

File: src/content/viral-amplicon/steps/Step3.md (hunk 4 above)
Comment (niemasd) [reply src/content/viral-amplicon/steps/Step3.md:47]: Yup, that's expected! That was the critical bug in v0.0.1: any read that *doesn't* have a gap between any two positions should support "no gap between these positions", which I now represent as an "empty-string" insertion. Without this, I was getting a bunch of spurious insertions when we had enough reads (e.g. 10) with the insertion purely by chance, even if we had thousands without the insertion

Comment (robertaboukhalil) [reply src/content/viral-amplicon/steps/Step3.md:47]: Perfect, thanks!

**Pull request merged**